src/cascadia/TerminalControl/ControlInteractivity.cpp

```diff
@@ -257,15 +256,15 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         }
         else if (WI_IsFlagSet(buttonState, MouseButtonState::IsRightButtonDown))
         {
-            // CopyOnSelect right click always pastes
-            if (_core->CopyOnSelect() || !_core->HasSelection())
+            // Try to copy the text and clear the selection

```

Comment:
It could be helpful to mention here that `CopySelectionToClipboard` returns `false` if the selection is empty, which results in our "unique" right-click behavior (copy if something's selected, paste if it isn't).